src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/ColorPalette.cs

```diff
@@ -16,33 +16,32 @@ public class ColorPaletteNodeViewCustomization : NotificationObject, INodeViewCu
         ///     WPF Control.
         /// </summary>
         private ColorPaletteUI ColorPaletteUINode;
+        private NodeView viewNode;
         private ColorPalette colorPaletteNode;
-        private Color mcolor;
-        /// <summary>
-        /// Selected Color
-        /// </summary>
-        public Color MColor
-        {
-            get { return mcolor; }
-            set
-            {                
-                mcolor = value;              
-                colorPaletteNode.dsColor = DSColor.ByARGB(mcolor.A, mcolor.R, mcolor.G, mcolor.B);
-                RaisePropertyChanged("MColor");
-            }
-        }
+        private Color color;
         /// <summary>
         ///     Customize View.
         /// </summary>
         /// <param name="model"></param>
         /// <param name="nodeView"></param>
         public void CustomizeView(ColorPalette model, NodeView nodeView)
         {
+            viewNode = nodeView;
             colorPaletteNode = model;
-            mcolor = Color.FromArgb(model.dsColor.Alpha, model.dsColor.Red, model.dsColor.Green, model.dsColor.Blue);
+            model.PropertyChanged += Model_PropertyChanged;
+            var undoRecorder = viewNode.ViewModel.WorkspaceViewModel.Model.UndoRecorder;
```

Comment:
any reason you are recording the state of the color picker immediately? When the user first selects a color - that should be the time you record the **old** color. Maybe this has to do with the two first undo bug you are seeing.